### PR TITLE
Migrate to poetry

### PR DESCRIPTION
**Pull request by hendzeld**
Migrate from Setuptools to Poetry

### CODE REVIEW

File: .github/workflows/test-run.yml [hunk 1]
```diff
@@ -90,6 +95,8 @@ jobs:
             ./tests/shieldhit/resources
 
   test-package:
+    env:
+      POETRY_VIRTUALENVS_CREATE: false
```
Comment (grzanka) [inline .github/workflows/test-run.yml:99]:
```suggestion
```

here we just install wheel package, I think poetry not needed here

File: README.md [hunk 2]
```diff
@@ -2,34 +2,26 @@
...
--   TOPAS
--   Fluka
-
-## Requirements
-
--   Python 3.9+
--   pip 21.3+
--   venv
+- SHIELD-HIT12A (beam.dat, mat.dat, geo.dat and detect.dat).
+- TOPAS
```
Comment (grzanka) [inline README.md:6]:
```suggestion
```

TOPAS not supported yet

**Comment by hendzeld** [comment]
@grzanka 

I tried to run this branch in UI.

Unfortunately, it doesn't work for now.

I read Pyodide documentation, and it can load public packages but only for `pure-Python` packages.
This means wheels with the `*-none-any.whl` suffix.
More here:
- https://pyodide.org/en/stable/usage/loading-packages.html
- https://pyodide.org/en/stable/usage/loading-packages.html#installing-packages

Unfortunately, SciPy doesn't provide such package.
It took me some time to realise that Pyodide has some packages already pre-build.
Old version used in UI `importScripts('https://cdn.jsdelivr.net/pyodide/v0.21.3/full/pyodide.js');` uses SciPy in version `1.8.1` according to this list: https://pyodide.org/en/0.21.3/usage/packages-in-pyodide.html#packages-built-in-pyodide

We can update Pyodide to the latest version (0.25.0) for now.
It uses SciPy in version `1.11.2`, according to: https://pyodide.org/en/0.25.0/usage/packages-in-pyodide.html

I will lock version of SciPy in converter for now to `1.11.2`

**Comment by grzanka** [comment]
@hendzeld another idea would be to get rid of scipy. It is being used only to handle rotation matrix. That could be moved to pure-python code.

**Comment by hendzeld** [comment]
> @hendzeld another idea would be to get rid of scipy. It is being used only to handle rotation matrix. That could be moved to pure-python code.

Maybe we should do this in in a different PR.

Regarding current version:
I tested this version on UI.
It seems to work, I will create a draft PR in the UI repository.

obraz

obraz

**Comment by grzanka** [comment]
> > @hendzeld another idea would be to get rid of scipy. It is being used only to handle rotation matrix. That could be moved to pure-python code.
> 
> Maybe we should do this in in a different PR.

Good, we can separate the issues.

Anyway thanks for deep inspection of the issue !

**Review by grzanka — APPROVED** [review]

**Pull request merged**